tests/testsuite/bad_config.rs

```diff
@@ -2151,6 +2151,56 @@ Caused by:
         .run();
 }
 
+#[cargo_test]
+fn github_pull_request_url() {
+    let p = project()
+        .file(
+            "Cargo.toml",
+            r#"
+                [package]
+                name = "foo"
+                version = "0.0.0"
+                edition = "2015"
+                authors = []
+
+                [dependencies.bar]
+                git = "https://github.com/foo/bar/pull/123"
+            "#,
+        )
+        .file("src/lib.rs", "")
+        .build();
+
+    p.cargo("check -v")
+        .with_status(101)
+        .with_stderr_data(str![[r#"
+[WARNING] dependency (bar) git url https://github.com/foo/bar/pull/123 is not a repository. The path looks like a pull request. Try replacing the dependency with: `git = "https://github.com/foo/bar.git" rev = "refs/pull/123/head"` in the dependency declaration.
+[UPDATING] git repository `https://github.com/foo/bar/pull/123`
+[WARNING] spurious network error (3 tries remaining): unexpected http status code: 404; class=Http (34)
+[WARNING] spurious network error (2 tries remaining): unexpected http status code: 404; class=Http (34)
+[WARNING] spurious network error (1 tries remaining): unexpected http status code: 404; class=Http (34)
+[ERROR] failed to get `bar` as a dependency of package `foo v0.0.0 ([ROOT]/foo)`
+
+Caused by:
+  failed to load source for dependency `bar`
+
+Caused by:
+  Unable to update https://github.com/foo/bar/pull/123
+
+Caused by:
+  failed to clone into: [ROOT]/home/.cargo/git/db/123-[HASH]
+
+Caused by:
+  network failure seems to have happened
+  if a proxy or similar is necessary `net.git-fetch-with-cli` may help here
+  https://doc.rust-lang.org/cargo/reference/config.html#netgit-fetch-with-cli
+
+Caused by:
+  unexpected http status code: 404; class=Http (34)
```

Comment:
Interesting. I thought it was network flakiness. Anyway, you can use `...` to omit the unimportant part.

```suggestion
...
[ERROR] failed to get `bar` as a dependency of package `foo v0.0.0 ([ROOT]/foo)`

Caused by:
  failed to load source for dependency `bar`

Caused by:
  Unable to update https://github.com/foo/bar/pull/123
...
```

Reply:
`[patch]` test also needs the same trick :)